File: Linked List/7. Linked List Cycle/solution.scala
/** Definition for singly-linked list. class ListNode(var _x: Int = 0) { var
  * next: ListNode = null var x: Int = _x }
  */

object Solution {
  def hasCycle(head: ListNode): Boolean = {
    if (head == null) return false
    var low = head
    var fast = head.next
    while (
      low != null && low.next != null && fast != null && fast.next != null
    ) {
      if (low == fast) return true
      low = low.next
      fast = fast.next.next
    }
    false
  }
}

File: Linked List/7. Linked List Cycle/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
	if head == nil {
		return false
	}
	low := head
	fast := head.Next
	for low != nil && low.Next != nil && fast != nil && fast.Next != nil {
		if low == fast {
			return true
		}
		low = low.Next
		fast = fast.Next.Next
	}
	return false
} /**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
	if head == nil {
		return false
	}
	low := head
	fast := head.Next
	for low != nil && low.Next != nil && fast != nil && fast.Next != nil {
		if low == fast {
			return true
		}
		low = low.Next
		fast = fast.Next.Next
	}
	return false
}
